### PR TITLE
SRCH-628 resolve deprecation warnings re. positional arguments

### PR DESCRIPTION
**Pull request by MothOnMars**
This PR resolves these deprecation warnings in our specs:
```
DEPRECATION WARNING: Using positional arguments in integration tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.
```

### CODE REVIEW

File: spec/requests/searches_spec.rb [hunk 1]
```diff
@@ -25,7 +28,9 @@
     context "when the request is from a mobile device" do
       before do
         iphone_user_agent = "Mozilla/5.0 (iPhone; U; CPU like Mac OS X; en) AppleWebKit/420+ (KHTML, like Gecko) Version/3.0 Mobile/1A543a Safari/419.3"
-        get "/search/docs", {:query => "pdf", :affiliate => affiliate.name}, { "HTTP_USER_AGENT" => iphone_user_agent }
+        get "/search/docs", params: { query: "pdf",
+                                      affiliate: affiliate.name },
```
Comment (peggles2) [inline spec/requests/searches_spec.rb]: make these single quotes

File: spec/requests/searches_spec.rb [hunk 2]
```diff
@@ -25,7 +28,9 @@
...
+                            headers:  { "HTTP_USER_AGENT" => iphone_user_agent }
       end
```
Comment (peggles2) [inline spec/requests/searches_spec.rb]: Can this be written headers: {  'HTTP_USER_AGENT': iphone_user_agent }

File: spec/requests/searches_spec.rb [hunk 3]
```diff
@@ -8,7 +8,10 @@
     context "when the request is from a mobile device" do
       before do
         iphone_user_agent = "Mozilla/5.0 (iPhone; U; CPU like Mac OS X; en) AppleWebKit/420+ (KHTML, like Gecko) Version/3.0 Mobile/1A543a Safari/419.3"
-        get "/search/news", {:query => 'element', :affiliate => affiliate.name, :channel => rss_feeds(:white_house_blog).id}, { "HTTP_USER_AGENT" => iphone_user_agent }
+        get "/search/news", params:  { query: 'element',
+                                       affiliate: affiliate.name,
+                                       channel: rss_feeds(:white_house_blog).id },
+                            headers: { "HTTP_USER_AGENT" => iphone_user_agent }
       end
```
Comment (peggles2) [inline spec/requests/searches_spec.rb]: single quotes here please

File: spec/requests/sayt_spec.rb [hunk 4]
```diff
@@ -22,24 +24,35 @@ def is_mobile_device?;
...
-      get '/sayt', :q => 'lorem  \\  ipsum', :callback => 'jsonp1234', :aid => affiliate.id
+      get '/sayt', params: { q: 'lorem  \\  ipsum',
+                             callback: 'jsonp1234',
+                             aid: affiliate.id }
       expect(response.body).to eq(%Q{/**/jsonp1234(#{phrases_in_json})})
     end
 
     it "should return empty JSONP if nothing matches the 'q' param string" do
-      get '/sayt', :q => "who moved my cheese", :callback => 'jsonp1276290049647', :aid => affiliate.id
+      get '/sayt', params: { q: "who moved my cheese",
```
Comment (peggles2) [inline spec/requests/sayt_spec.rb]: single quotes

File: spec/requests/searches_spec.rb (hunk 3 above)
Comment (peggles2) [reply spec/requests/searches_spec.rb]: Can this be written headers: { 'HTTP_USER_AGENT': iphone_user_agent }

File: spec/requests/searches_spec.rb (hunk 2 above)
Comment (MothOnMars) [reply spec/requests/searches_spec.rb]: I'll switch to single quotes, but we should keep these as string keys, as that's the standard format for headers:
https://api.rubyonrails.org/classes/ActionDispatch/Http/Headers.html

File: spec/requests/searches_spec.rb (hunk 1 above)
Comment (MothOnMars) [reply spec/requests/searches_spec.rb]: 👍

File: spec/requests/clicked_spec.rb [hunk 5]
```diff
@@ -17,21 +17,44 @@
...
+                                s: @module,
+                                v: @vertical,
+                                l: @locale,
+                                i: @model_id }
       expect(response.success?).to be(true)
       expect(response.body).to eq('')
     end
 
     it "should log the click" do
       expect(Click).to receive(:log).with(@unescaped_url, @query, '2010-04-22 23:28:25', '127.0.0.1', @affiliate_name, @position, @module, @vertical, @locale, anything(), @model_id)
```
Comment (peggles2) [inline spec/requests/clicked_spec.rb]: while you are at it can you take should off from here too?

**Pull request merged**